pkg/codegen/schema.go

```diff
@@ -169,8 +169,9 @@ func GenerateGoSchema(sref *openapi3.SchemaRef, path []string) (Schema, error) {
 		outSchema.GoType = "interface{}"
 		return outSchema, nil
 	}
-	// We can't support this in any meaningful way
-	if schema.OneOf != nil {
+	// We can't support this in any meaningful way,
+	// except if oneOf defines just restriction on properties via the "required" property
+	if schema.OneOf != nil && !(schema.OneOf[0].Value != nil && schema.OneOf[0].Value.Required != nil) {
```

Comment:
This logic only looks at the first item in the `OneOf` slice. What if there are more items in the slice and they're not all "required"?

Reply:
In such a case it wouldn't be correct. The assumption here is that purpose of `oneOf` is that all items are of the same kind (type, restriction). 
Sure, in order to be completely correct all items should be checked.

Reply:
Given that this is dealing with arbitrary OpenAPI specs, ISTM that that assumption should be checked in the code, otherwise this probably isn't a reasonable thing to accept into master.

Reply:
One other thing: this will panic if `schema.OneOf` is non-nil but zero length.